client/components/ProductSingle.js

```diff
@@ -1,37 +1,37 @@
 /* eslint-disable react/destructuring-assignment */
+/* eslint-disable no-shadow */
 import React from 'react';
-import axios from 'axios';
-import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { fetchProductReviews, fetchProduct } from '../store';
 
-class ProductSingle extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      product: {},
-      reviews: [],
-    };
-  }
-
-  componentDidMount() {
-    // TODO: what's a more efficient way of doing this (that linter likes)?
-    const { match } = this.props;
-    const { params } = match;
-    const { productId } = params;
+const mapStateToProps = (state, props) => {
+  const { products } = state.product;
```

Comment:
See my comments in `store` about creating a `product` reducer, along with action type and action creator.

```suggestion
  const { product } = state.product;
```